src/App.js

```diff
@@ -1,24 +1,62 @@
 import logo from './logo.svg';
 import './App.css';
+import React from 'react';
+import { useState } from 'react';
+import { useEffect } from 'react';
+import '@fontsource/roboto/300.css';
+import Header from './components/Header';
+import Article from './components/Article';
+import { ThemeProvider, createTheme } from '@mui/material/styles';
+
+
+const theme = createTheme({
+  palette: {
+    primary: {
+      main: '#3d5af1',
+    },
+    secondary: {
+      main:'#f1d33d', 
+    },
+    background: {
+      main: '#fff'
+    }
+  },
+});
 
 function App() {
+
+  const [articles, setArticles] = useState([]);
+  const [subreddit, setSubreddit] = useState('');
+
+  useEffect(()=> {
+    fetch('https://www.reddit.com/'+subreddit+'.json').then(res=>{
+      if (res.status !=200) {
+        console.log('Error');
+        return;
+      }
+
+      res.json().then(data => {
+        if (data != null) {
+          setArticles(data.data.children);
+        }
+      })
```

Comment:
if we have "data" inside children, we should do : data.data.children.data to setArticles here.

Reply:
I tried to do this but it didnt work so i keep it like this!!!